### PR TITLE
SEO optimization

### LINKED ISSUE #344: Optimizing SEO - Very important!
Optimize the website SEO so that you can e.g. can also be found better in Google, Bing and so on

### Things that need to be adjusted
- Keywords are completely missing. 
- In many blog posts it doesn't have a description or says author.
- Important meta tags for the search engine are missing.

### Infromation about SEO
- [https://developers.google.com/search/docs/beginner/seo-starter-guide](https://developers.google.com/search/docs/beginner/seo-starter-guide)
- [https://www.semrush.com/blog/seo-tips/](https://www.semrush.com/blog/seo-tips/)

### Tools
- [https://de.semrush.com/analytics/keywordmagic/start](https://de.semrush.com/analytics/keywordmagic/start)
- [https://www.keywordtooldominator.com/](https://www.keywordtooldominator.com/)
- [https://www.ionos.de/digitalguide/websites/web-entwicklung/die-wichtigsten-meta-tags-im-ueberblick/](https://www.ionos.de/digitalguide/websites/web-entwicklung/die-wichtigsten-meta-tags-im-ueberblick/)

### PR DESCRIPTION
**Pull request by ClientCrash**
**Issue**
Closes #344

### CODE REVIEW

File: pages/blog/_slug.vue [hunk 1]
```diff
@@ -41,7 +41,12 @@ export default {
       const response = await fetch('https://staging-admin-api.cryptic-game.net/website/blog/en/' + this.slug)
       post = await response.json()
     }
-
+    // eslint-disable-next-line eqeqeq
+    if (post.author == undefined) {
```
Comment (MarcelCoding) [inline pages/blog/_slug.vue:45]:
```suggestion
    if (!post.author) {
```

File: pages/blog/_slug.vue [hunk 2]
```diff
@@ -41,7 +41,12 @@ export default {
...
+      post.author = 'Unknown'
+    }
+    // eslint-disable-next-line no-console
+    console.log(post)
```
Comment (MarcelCoding) [inline pages/blog/_slug.vue:49]:
```suggestion
```

File: pages/blog/_slug.vue [hunk 3]
```diff
@@ -66,7 +71,8 @@ export default {
           hid: 'article:published_time',
           name: 'article:published_time',
           content: new Date(this.post.created).toString()
-        }
+        },
+        { hid: 'author', name: 'author', content: this.post.author }
```
Comment (MarcelCoding) [inline pages/blog/_slug.vue:75]: Maybe don't use author unknown. Just don't include this meta tag.

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

**Comment by EggerWebdesign** [comment]
Keywords are missing

**Comment by MarcelCoding** [comment]
@EggerWebdesign https://github.com/MarcelCoding/m4rc3l.de/blob/main/pages/blog/_slug.vue#L109-L130 what about this method? cc. @ClientCrash

**Comment by MarcelCoding** [comment]
1. replace special chars with spaces
2. split on spaces
3. to lowercase
4. count how often every word is used
5. find the top 15
6. return the top 15 joined by `, `

**Comment by Webkinger** [comment]
Currently you can do it this way but later I would still make that you can edit/type the keywords Manuel in the admin panel.

**Review by MarcelCoding — APPROVED** [review]

**Comment by MarcelCoding** [comment]
Are the keywords still relevant or can this be merged?

**Comment by Webkinger** [comment]
Seo technically, keywords are important

**Comment by MarcelCoding** [comment]
@ClientCrash do you want to implement keywords?

**Review by MarcelCoding — APPROVED** [review]

**Review by DevMimas — APPROVED** [review]

**Pull request merged**